docs/pages/access-controls/guides/locking.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: Locking
+description: Temporarily lock out specific access pathways.
+---
+
+# Session & Identity Locking
+
+Teleport provides a locking mechanism to restrict access to a computing
+environment on an *ad hoc* basis, e.g. to disable a compromised user or
+prevent access during cluster maintenance.
+
+When a lock is in force, all interactions (like SSH/DB/k8s connections or
+certificate requests) matched by the lock's target are rejected.
+A lock can target the following objects or attributes:
+- a Teleport user by the user's name;
+- a Teleport [RBAC](../reference.mdx) role by the role's name;
+- an MFA device by the device's UUID;
+- an OS/UNIX login;
+- a Teleport node by the node's UUID (effectively unregistering it from the
+  cluster).
+
+<Admonition type="warning">
+  Locking support was introduced in Teleport v7.1. To fully enforce the locks
+  at all components, you should update **all** `teleport` binaries in your
+  deployment.
+</Admonition>
+
+## Creating a new lock
+
+Locks are modeled as [resources](../../admin-guide.mdx#resources) with `kind:
+lock`.  To easily create a new lock, one can run the `tctl lock` command:
+
+```
+$ tctl lock --user=foo@example.com --message="Suspicious activity." --ttl=10h
+Created a lock with name "dc7cee9d-fe5e-4534-a90d-db770f0234a1".
+```
+
+Note that without specifying `--ttl` or `--expires` the created lock remains in
+force until explicitly removed with `tctl rm`.  Refer to `tctl lock --help` for
+the list of all supported parameters.
+
+Under the hood, `tctl lock` creates a resource similar to the following:
+
+```yaml
+kind: lock
+version: v2
+metadata:
+  name: dc7cee9d-fe5e-4534-a90d-db770f0234a1
+spec:
+  target:
+    user: foo@example.com
+  message: "Suspicious activity."
+  expires: "2021-08-14T22:27:00Z"  # RFC3339 format
+```
+
+The `kind: lock` resources can also be created and updated using `tctl create`
+as per usual, see the [Admin Guide](../../admin-guide.mdx#resources) for more
+details.
+
+## Lock in force
+
+With a lock in force, all established connections involving the lock's target
+get terminated while any new requests are rejected.
+
+Errors returned and warnings logged in this situation feature a message of the
+form:
+```
+lock targeting User:"foo@example.com" is in force: Suspicious activity.
+```
+
+## Locking mode
+
+If a Teleport component cannot properly synchronize its local lock view with
+the backend, there is a decision to be made about whether to rely on the last
+known locks. This decision strategy is encoded as one of the two modes:
+- `strict` mode causes all interactions to be terminated when the locks are not
+  guaranteed to be up to date;
+- `best_effort` mode keeps relying on the most recent locks.
+
+The cluster-wide mode defaults to `best_effort` and can be set in
+`teleport.yaml` of the Auth server:
+
+```yaml
```

Comment:
Can this be done via auth_preference resource as well?

Reply:
Yes, though that configuration method doesn't seem to be documented anywhere so I followed the convention of the rest of the documentation to show just the `teleport.yaml` snippet.